code/clue.js

```diff
@@ -1,59 +1,396 @@
+//First param: type of event. Second param: the function to call on event.
+document.getElementById('killerCard').addEventListener('click', pickKiller);
+
+//If pickKiller function is excecuted/card is clicked then excecute this function:
+const pickWeapon = () => {
+  if (killerIsPicked && !weaponIsPicked) {
+    document.getElementById('weaponLoader').style.visibility='visible'
+    setTimeout(function () {
+      document.getElementById('weaponLoader').style.visibility='hidden'
+      mystery.weapon = randomSelector(weapons); //Randomly select a weapon from weapons and add it to the mystery object
+      document.getElementById('weaponCard').style.background = mystery.weapon.color
+  document.getElementById('weaponName').innerHTML = `${mystery.weapon.name}`
+  document.getElementById('weaponWeight').innerHTML = `Weight: ${mystery.weapon.weight}`
+  document.getElementById('weaponFlexibility').innerHTML = `Flexibility: ${mystery.weapon.flexibility}`
+  document.getElementById('weaponImage').src = `${mystery.weapon.image}`
+  weaponIsPicked = true
+    }, 2500)
+  }
+};
+document.getElementById('weaponCard').addEventListener('click', pickWeapon);
+
+//If pickKiller AND pickWeapon function is excecuted/cards are clicked then excecute this function on the pickroom:
+const pickRoom = () => {
+  if (killerIsPicked && weaponIsPicked) {
+    document.getElementById('roomLoader').style.visibility='visible'
+    document.getElementById('revealButton').style.visibility='visible'//Makes the reveal-button visible when the room-card has been clicked.
+    setTimeout(function () {
+      document.getElementById('roomLoader').style.visibility='hidden'
+      mystery.room = randomSelector(rooms);//Randomly select a room from rooms and add it to the mystery object
+      document.getElementById('roomName').innerHTML = `${mystery.room.name}`
+      document.getElementById('roomImage').src = `${mystery.room.image}`
+      roomIsPicked = true
+    }, 2500)
+  }
+};
+document.getElementById('roomCard').addEventListener('click', pickRoom); 
+
+// STEP 4 - CREATE A FUNCTION revealMystery that will be invoked when you click that button. 
+const revealMystery = (mystery) => {
+  //Invoke when button is clicked and show elements from mystery object in text:
+document.getElementById('mystery').innerHTML = `The murder was committed by ${mystery.killer.firstName} in the ${mystery.room.name} with ${mystery.weapon.name}` 
+document.getElementById('revealButton').innerHTML = `PLAY AGAIN?` //change the text on button.
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+//BELOW I TRIED TO MAKE A FUNCTION SO THAT THE PAGE RELOADS WHEN 'PLAY AGAIN'-BUTTON IS CLICKED. IT DIDN'T WORK. I wanted to leave this here just to show the code-reviewer what I tried and maybe get pointers on how to achieve this.
```

Comment:
You had some good attempts here, but we can use your logic without adding a separate `onclick` handler

```javascript
const revealMystery = (mystery) => {
  // If the text is "PLAY AGAIN?" then the mystery has already been revealed
  if (document.getElementById('revealButton').innerText === 'PLAY AGAIN?') {
    location.reload();
  } else {
  // The text has not been changed yet, so we are revealing the mystery
    document.getElementById(
      'mystery'
    ).innerHTML = `The murder was committed by ${mystery.killer.firstName} in the ${mystery.room.name} with ${mystery.weapon.name}`;

    // Update the button text to reflect the action that can now be performed 
    document.getElementById('revealButton').innerText = 'PLAY AGAIN?'; //change the text on button.
  }
};
```

Reply:
Thank you! Now it is working :)